sklearn/base.py

```diff
@@ -449,6 +449,25 @@ def fit_transform(self, X, y=None, **fit_params):
             return self.fit(X, y, **fit_params).transform(X)
 
 
+class DensityMixin(object):
+    """Mixin class for all density estimators in scikit-learn."""
+    _estimator_type = "DensityEstimator"
+
+    def score(self, X, y = None):
+        """Returns the log likelihood under the model on the given data X.
+
+        Parameters
+        ----------
+        X : array-like, shape = (n_samples, n_features)
+
+        Returns
+        -------
+        score: float
+            sum of log probability of samples
+        """
+        return np.sum(self.score_samples(X))
```

Comment:
likelihood is often referred as a function of parameters. For individual data samples, I think we should call the same numerical value as log probabilities.